ksql-execution/src/main/java/io/confluent/ksql/execution/materialization/MaterializationInfo.java

```diff
@@ -0,0 +1,196 @@
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.execution.materialization;
+
+import static java.util.Objects.requireNonNull;
+
+import com.google.errorprone.annotations.Immutable;
+import io.confluent.ksql.execution.expression.tree.Expression;
+import io.confluent.ksql.execution.plan.SelectExpression;
+import io.confluent.ksql.schema.ksql.LogicalSchema;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Pojo for passing around information about materialization of a query's state store
+ */
+@Immutable
+public final class MaterializationInfo {
+
+  private final String stateStoreName;
+  private final LogicalSchema stateStoreSchema;
+  private final List<TransformInfo> transforms;
+  private final LogicalSchema schema;
+
+  /**
+   * Create instance.
+   *
+   * @param stateStoreName the name of the state store
+   * @param stateStoreSchema the schema of the data in the state store
+   * @param transforms a list of transformations to be applied to the data in the store
+   * @param schema the schema of the result of applying transformations to the data in the store
+   * @return instance.
+   */
+  public static MaterializationInfo of(
+      final String stateStoreName,
+      final LogicalSchema stateStoreSchema,
+      final List<TransformInfo> transforms,
+      final LogicalSchema schema
+  ) {
+    return new MaterializationInfo(stateStoreName, stateStoreSchema, transforms, schema);
+  }
+
+  public String stateStoreName() {
+    return stateStoreName;
+  }
+
+  public LogicalSchema getStateStoreSchema() {
+    return stateStoreSchema;
+  }
+
+  public LogicalSchema getSchema() {
+    return schema;
+  }
+
+  public List<TransformInfo> getTransforms() {
+    return transforms;
+  }
+
+  private MaterializationInfo(
+      final String stateStoreName,
+      final LogicalSchema stateStoreSchema,
+      final List<TransformInfo> transforms,
+      final LogicalSchema schema
+  ) {
+    this.stateStoreName = requireNonNull(stateStoreName, "stateStoreName");
+    this.stateStoreSchema = requireNonNull(stateStoreSchema, "stateStoreSchema");
+    this.transforms = requireNonNull(transforms, "transforms");
+    this.schema = requireNonNull(schema, "schema");
+  }
+
+  public static class Builder {
+    private final String stateStoreName;
+    private final LogicalSchema stateStoreSchema;
+    private final List<TransformInfo> transforms;
+    private LogicalSchema schema;
+
+    public Builder(final String stateStoreName, final LogicalSchema stateStoreSchema) {
+      this.stateStoreName = Objects.requireNonNull(stateStoreName, "stateStoreName");
+      this.stateStoreSchema = Objects.requireNonNull(stateStoreSchema, "stateStoreSchema");
+      this.transforms = new LinkedList<>();
+      this.schema = stateStoreSchema;
+    }
+
+    public Builder mapAggregates(final AggregatesInfo aggregatesInfo, final LogicalSchema schema) {
```

Comment:
renaming `schema` param to something that doesn't clash with the `schema` field would make this code more readable, e.g.

```suggestion
    public Builder mapAggregates(final AggregatesInfo aggregatesInfo, final LogicalSchema aggregatedSchema) {
```